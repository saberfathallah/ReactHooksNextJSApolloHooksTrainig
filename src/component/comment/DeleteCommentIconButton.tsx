import React from "react";

import IconButton from "@material-ui/core/IconButton";
import DeleteIcon from "@material-ui/icons/Delete";

interface VariablesDelete {
  variables: {
    deleteCommentInput: {
      commentId: string;
      postId: string;
    };
  };
}

interface EditCommentIconButton {
  deleteComment: (variables: VariablesDelete) => any;
  commentId: string;
  postId: string;
}

const EditCommentIconButton = ({
  deleteComment,
  postId,
  commentId,
}: EditCommentIconButton) => (
  <IconButton
    aria-label="delete"
    onClick={(): Promise<object> =>
      deleteComment({
        variables: {
          deleteCommentInput: { postId, commentId },
        },
      })
    }
  >
    <DeleteIcon fontSize="small" />
  </IconButton>
);

export default EditCommentIconButton;
