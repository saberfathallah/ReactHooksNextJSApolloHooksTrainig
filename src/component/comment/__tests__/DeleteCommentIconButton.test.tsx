import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import DeleteCommentIconButton from "../DeleteCommentIconButton";

describe("DeleteCommentIconButton", () => {
  it("should match snapshot", () => {
    const tree = renderer.create(
      <DeleteCommentIconButton
        deleteComment={() => {}}
        postId="postId"
        commentId="commentId"
      />
    );
    expect(tree).toMatchSnapshot();
  });

  it("shouldn't call addLike should call dislike", () => {
    const deleteComment = jest.fn();
    const wrapper = mount(
      <DeleteCommentIconButton
        deleteComment={deleteComment}
        postId="postId"
        commentId="commentId"
      />
    );
    wrapper.find("button").simulate("click");
    expect(deleteComment).toHaveBeenCalledTimes(1);
  });
});
