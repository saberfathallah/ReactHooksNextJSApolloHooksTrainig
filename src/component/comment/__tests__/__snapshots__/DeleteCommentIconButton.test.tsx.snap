// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteCommentIconButton should match snapshot 1`] = `
<button
  aria-label="delete"
  className="MuiButtonBase-root MuiIconButton-root"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiIconButton-label"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
      />
    </svg>
  </span>
</button>
`;
